networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PNetwork.java

```diff
@@ -224,15 +223,7 @@ private Gossip createGossip() {
 
   private Multiaddr getAdvertisedAddr(NetworkConfig config, final NodeId nodeId) {
     try {
-      final InetSocketAddress advertisedAddress =
-          new InetSocketAddress(config.getAdvertisedIp(), config.getAdvertisedPort());
-      final InetSocketAddress resolvedAddress;
-      if (advertisedAddress.getAddress().isAnyLocalAddress()) {
-        resolvedAddress =
-            new InetSocketAddress(InetAddress.getLocalHost(), advertisedAddress.getPort());
-      } else {
-        resolvedAddress = advertisedAddress;
-      }
+      final InetSocketAddress resolvedAddress = MultiaddrUtil.getResolvedInetSocketAddress(config);
```

Comment:
I'm not convinced this code belongs in `MultiaddrUtil` - it doesn't actually have anything to do with Multiaddres (just our config and InetSocketAddress).  It wouldn't need to move if we're getting the discv5 address from the local node record.